.github/workflows/everything.yml

```diff
@@ -218,16 +279,34 @@ jobs:
           path: source
       - name: Setup
         run: gha/scripts/ci/gh-actions/macos-setup.sh
+      - name: Restore cache
+        uses: actions/cache/restore@v3
+        id: restore-cache
+        with:
+          path: .ccache
+          key: ccache-${{ matrix.os }}-${{ matrix.compiler }}-${{ matrix.parallel }}-${{ github.sha }}
+          restore-keys: |
+            ccache-${{ matrix.os }}-${{ matrix.compiler }}-${{ matrix.parallel }}
+      - name: Configure cache
+        run: ccache -z
       - name: Update
         run: gha/scripts/ci/gh-actions/run.sh update
       - name: Configure
         run: gha/scripts/ci/gh-actions/run.sh configure
       - name: Build
         run: gha/scripts/ci/gh-actions/run.sh build
+      - name: Print ccache statistics
+        run: ccache -s
+      - name: Save cache
+        uses: actions/cache/save@v3
+        if: github.ref_name == 'master' && steps.restore-cache.outputs.cache-hit == 'false'
```

Comment:
Does this mean only save if there was no cache hit?

Reply:
Yes, if there was an "exact match" on the cache key (which is defined using the sha of the merge commit getting tested), then attempting to write the cache would produce the exact same contents.  Since github caches are not editable (updatable), it would be an error to write it anyway, so I'm avoiding.